lib/actions/model.js

```diff
@@ -45,8 +46,14 @@ module.exports = {
       if (preSaveAction) dispatch(preSaveAction);
 
       const Model = storageManager.storage(brainstemKey).model;
-      const xhr = new Model({ id: modelId })
-        .save(attributes, saveOptions);
+      const model = new Model({ id: modelId });
+
+      if (!model.isValid()) {
+        dispatch(invalidAction(model.validationError));
+        return model.validationError;
+      }
+
+      const xhr = model.save(attributes, saveOptions);
```

Comment:
Yeah. Just to regurgitate: if `false` then return the validation errors attached to the model else return the promise.